helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -357,6 +358,12 @@ void stopPeriodRebalance() {
     }
   }
 
+  private void stopOnDemandTimer() {
```

Comment:
nit, let's call this shutdownOnDemandTimer.

Considering the periodRebalanceTimer can be started and stopped repeatedly. But this shutdownOnDemandTimer will be one-time thing in our logic.

Reply:
changed.